edward/data.py

```diff
@@ -30,57 +38,71 @@ def __init__(self, data=None, shuffled=True):
             # shuffle self.data
             raise NotImplementedError()
 
-        self.counter = 0
         if self.data is None:
-            self.N = None
+            pass
         elif isinstance(self.data, tf.Tensor):
             self.N = self.data.get_shape()[0].value
+            self.counter = 0
```

Comment:
@dustinvtran can you document what counter is? i don't intuitively expect a counter in a data class, and it doesn't pass the 60 second test. (i'm still trying to parse what the counter is doing.)